### PR TITLE
Testing/sleep-repository

### LINKED ISSUE #22: Build tests for SleepRepository class
**Related Problem(s)**
Test properties and methods of `SleepRepository`, which handles all calculations that use multiple users' sleep data

### PR DESCRIPTION
**Pull request by richardltyler**
#### Description
- add test to guide build of `SleepRepository` class

#### Functionality area
- styling
- html layout
- **data model**
- DOM
- data model-DOM interaction
- documentation

#### Where a reviewer should start
- SleepRepository-test.js

#### How to test this change
- run `npm test test/SleepRepository-test.js` in terminal

#### Next Steps/Loose Ends
- build SleepRepository class

### CODE REVIEW

**Comment by richardltyler** [comment]
I added in some commented out code. It will break if you try to run the tests without a sleep repository class to push into it. Make sure to remove those later!

File: test/SleepRepository-test.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+const { expect } = require('chai');
+const chai = require('chai');
+const expecet = chai.expect;
```
Comment (kellydinneen) [inline test/SleepRepository-test.js:3]: `expect`

File: test/SleepRepository-test.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        "date": "2019/06/21",
+        "hoursSlept": 6.5,
+        "sleepQuality": 4.0
+      }
+    ];
+    // sleepRepo = new SleepRepository(sleepData);
+  });
+
+  it.skip('should be a function', function() {
+    expect(SleepRepository).to.be.be.a('function');
```
Comment (kellydinneen) [inline test/SleepRepository-test.js:73]: extra `be`

File: test/SleepRepository-test.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  });
+
+  it.skip('should instantiate the Sleep Repository', function() {
+    expect(sleepRepo).to.be.an.instanceOf(SleepRepository);
+  });
+
+  it.skip('should hold all of the sleep objects', function() {
```
Comment (kellydinneen) [inline test/SleepRepository-test.js:80]: Ok, so its a straightforward copy of the raw data!!! If the class methods work out easily, I should probably revise the `ActivityRepository` tests, which assume some fancier footwork when processing the data in the constructor

File: test/SleepRepository-test.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    expect(sleepRepo.sleep).to.deep.equal(sleepData);
+  });
+
+  it.skip('should find all user\'s average sleep quality', function() {
+    const averageQuality = sleepRepo.calculateAllUsersAverageSleepQuality();
+    expect(averageQuality).to.equal(2.9);
```
Comment (kellydinneen) [inline test/SleepRepository-test.js:86]: ok -- so method will have to limit decimal places unless 2.9 is exact

**Review by kellydinneen — APPROVED** [review]
Looks good besides a few typos, which I can fix while I'm working on the class.

I am thinking about whether or not it may be worth processing the `sleepData` just a bit more before it gets gobbled by the methods. I'll play it by ear.

**Pull request merged**